src/entity.ts

```diff
@@ -302,19 +303,71 @@ export namespace entity {
    *   stringValue: 'Hi'
    * });
    * // 'Hi'
+   * decodeValueProto({
+   *   stringValue: '1',
+   *   valueType: 'stringValue',
+   *   name:'isActive'
+   * },{
+   *         typeCast: function(field, next) {
+   *             if (field.valueType === 'stringValue' && field.name==='isActive') {
+   *                 return field[field.valueType] === '1';
+   *             } else {
+   *                 return next()
+   *             }
+   *         }
+   *   }
+   * );
+   * // true
    *
    * decodeValueProto({
    *   blobValue: Buffer.from('68656c6c6f')
    * });
    * // <Buffer 68 65 6c 6c 6f>
    */
-  export function decodeValueProto(valueProto: ValueProto) {
+  export function decodeValueProto(
+    valueProto: ValueProto,
+    typeCast?: Function
+  ) {
+    if (typeof typeCast === 'function') {
+      return typeCast!(valueProto, () => {
+        return decodeValue(valueProto, typeCast);
+      });
+    } else {
+      return decodeValue(valueProto, typeCast);
+    }
+  }
+  /**
+   * Convert a protobuf Value message to its native value without type casting.
+   *
+   * @private
+   * @param {object} valueProto The protobuf Value message to convert.
+   * @param {function} [typeCast] The typeCast function to handle type casting for array and object.
+   * @returns {*}
+   *
+   * @example
+   * decodeValue({
+   *   booleanValue: false
+   * });
+   * // false
+   *
+   * decodeValue({
+   *   stringValue: 'Hi'
+   * });
+   * // 'Hi'
+   * decodeValue({
+   *   blobValue: Buffer.from('68656c6c6f')
+   * });
+   * // <Buffer 68 65 6c 6c 6f>
+   */
+  function decodeValue(valueProto: ValueProto, typeCast?: Function) {
```

Comment:
Why introduce new function,
Can't `decodeValueProto` be overloaded?

Reply:
because it repeated code used in  `decodeValueProto` function, otherwise I need to repeat while code.